sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/aio/change_feed_fetcher.py

```diff
@@ -0,0 +1,206 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Internal class for processing change feed implementation in the Azure Cosmos
+database service.
+"""
+import base64
+import copy
+import json
+from abc import ABC, abstractmethod
+from typing import Dict, Any, List
+
+from azure.cosmos import http_constants, exceptions
+from azure.cosmos._change_feed.change_feed_start_from import ChangeFeedStartFromPointInTime
+from azure.cosmos._change_feed.change_feed_state import ChangeFeedStateV1, ChangeFeedStateV2
+from azure.cosmos.aio import _retry_utility_async
+from azure.cosmos.exceptions import CosmosHttpResponseError
+
+# pylint: disable=protected-access
+
+class ChangeFeedFetcher(ABC):
+
+    @abstractmethod
+    async def fetch_next_block(self):
+        pass
+
+class ChangeFeedFetcherV1(ChangeFeedFetcher):
+    """Internal class for change feed fetch v1 implementation.
+     This is used when partition key range id is used or when the supplied continuation token is in just simple etag.
+     Please note v1 does not support split or merge.
+
+    """
+    def __init__(
+            self,
+            client,
+            resource_link: str,
+            feed_options: Dict[str, Any],
+            fetch_function):
+
+        self._client = client
+        self._feed_options = feed_options
+
+        self._change_feed_state = self._feed_options.pop("changeFeedState")
+        if not isinstance(self._change_feed_state, ChangeFeedStateV1):
+            raise ValueError(f"ChangeFeedFetcherV1 can not handle change feed state version"
+                             f" {type(self._change_feed_state)}")
+
+        self._resource_link = resource_link
+        self._fetch_function = fetch_function
+
+    async def fetch_next_block(self):
+        """Returns a block of results.
+
+        :return: List of results.
+        :rtype: list
+        """
+        async def callback():
+            return await self.fetch_change_feed_items(self._fetch_function)
+
+        return await _retry_utility_async.ExecuteAsync(self._client, self._client._global_endpoint_manager, callback)
+
+    async def fetch_change_feed_items(self, fetch_function) -> List[Dict[str, Any]]:
```

Comment:
Can the abstract definition of `fetch_change_feed_items` be moved to `ChangeFeedFetcher` - the abstract class?

Reply:
the implementation is different between V1 and V2 for the fetch_change_feed_items (mostly around how to handle the continuation token)